go-controller/pkg/cni/cni.go

```diff
@@ -174,9 +174,29 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(kubeAuth *KubeAPIAuth, clientse
 			if err != nil {
 				return nil, err
 			}
-			if _, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo); err != nil {
+			primaryNetRes, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo)
+			if err != nil {
 				return nil, err
 			}
+
+			originalNumberOfIfaces := len(response.Result.Interfaces)
+			response.Result.Interfaces = append(response.Result.Interfaces, primaryNetRes.Interfaces...)
+			for i := range primaryNetRes.IPs {
+				ifaceIPConfig := primaryNetRes.IPs[i].Copy()
+
+				// we add a single ovn-udn1 primary interface, and must account
+				// for the host veths, which we also return.
+				// The order matters, hence the + 1 below. Check how we return
+				// the interfaces in
+				// https://github.com/ovn-org/ovn-kubernetes/blob/a1e88dcd0b79b6c5eb033daea178708993f8664e/go-controller/pkg/cni/helper_linux.go#L650
+				newIDX := originalNumberOfIfaces + 1
```

Comment:
Shouldn't this be offset based? Like:
```
newIDX := ifaceIPConfig.Interface + originalNumberOfIfaces
```

Have you tested when you get multiple IPs for the same interface, for example, dual-stack?

Reply:
> Have you tested when you get multiple IPs for the same interface, for example, dual-stack?

Yes, I did. Works as expected. If you want, I can update the yamls on the PR description to use a dual-stack network instead.

For this NAD:
```yaml
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: access-tenant-blue
  namespace: default
spec:
  config: |2

    {
            "cniVersion": "0.3.0",
            "name": "tenantblue",
            "type": "ovn-k8s-cni-overlay",
            "topology":"layer2",
            "subnets": "10.128.0.0/16,fd10:1234::0/64",
            "mtu": 1300,
            "netAttachDefName": "default/access-tenant-blue",
            "role": "primary"
    }
```

It returns this annotation:
```sh
kubectl get pods pod -ojsonpath="{@.metadata.annotations.k8s\.v1\.cni\.cncf\.io\/network-status}" | jq
[
  {
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
      "10.244.2.3",
      "fd00:10:244:3::3"
    ],
    "mac": "0a:58:0a:f4:02:03",
    "default": true,
    "dns": {}
  },
  {
    "name": "ovn-kubernetes",
    "interface": "ovn-udn1",
    "ips": [
      "10.128.0.4",
      "fd10:1234::4"
    ],
    "mac": "0a:58:0a:80:00:04",
    "default": true,
    "dns": {}
  }
]
```

The pod's IPs:
```sh
kubectl exec pod -- ip a show ovn-udn1
3: ovn-udn1@if12: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1300 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:80:00:04 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.128.0.4/16 brd 10.128.255.255 scope global ovn-udn1
       valid_lft forever preferred_lft forever
    inet6 fd10:1234::4/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::858:aff:fe80:4/64 scope link 
       valid_lft forever preferred_lft forever
```

> Shouldn't this be offset based?

Good point ! It reads a lot better IMHO. Let me just check it works as expected, and I'll re-push.

Reply:
Done.